### PR TITLE
AC test client

### PR DESCRIPTION
**Pull request by sethalves**
- added a command-line tool for checking on health of a domain-server and ACs

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 35755ddcb1f96d6ca433b98fbb081048484a60c0 available at: https://gist.github.com/a2eca4289f5f4864cb745f28217f7e33<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-35755dd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-35755dd.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1dc52d3bf579d6d621fc9f3109427c5a566ffed2 available at: https://gist.github.com/a2eca4289f5f4864cb745f28217f7e33<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-1dc52d3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-1dc52d3.dmg

**Comment by sethalves** [comment]
to test:
- build on Linux and run `./tools/ac-client/ac-client -v alpha.highfidelity.io`
- look for "success" near the end of the output

**Comment by sethalves** [comment]
Here's how it can be used in a checking script:
    ./tools/ac-client/ac-client -d eschatology.hungry.com 2>&1 | while read I; do echo "failed:" $I; done

**Comment by hifi-gustavo** [comment]
Build console output for commit: 86a31903851c2ee9b82933aa6ba12d7ca124e54c available at: https://gist.github.com/a2eca4289f5f4864cb745f28217f7e33<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-86a3190.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-86a3190.dmg

File: tools/ac-client/src/ACClientApp.cpp [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+#include "ACClientApp.h"
+
+ACClientApp::ACClientApp(int argc, char* argv[]) :
+    QCoreApplication(argc, argv)
+{
+    // parse command-line
+    QCommandLineParser parser;
+    parser.setApplicationDescription("High Fidelity AC client");
+    parser.addHelpOption();
```
Comment (zzmp) [inline tools/ac-client/src/ACClientApp.cpp]: Isn't this redundant with L32?

Comment (sethalves) [inline tools/ac-client/src/ACClientApp.cpp]: yeah

**Comment by hifi-gustavo** [comment]
Build console output for commit: 785d67b87a6377813d64b43c1946facac092c795 available at: https://gist.github.com/a2eca4289f5f4864cb745f28217f7e33<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-785d67b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-785d67b.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f338e38bcf7eaeeae290ae721aa5d49275163c42 available at: https://gist.github.com/a2eca4289f5f4864cb745f28217f7e33<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-f338e38.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-f338e38.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0510411ac139fb963d6ad25900b11c88296c2b24 available at: https://gist.github.com/a2eca4289f5f4864cb745f28217f7e33<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-0510411.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-0510411.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7b5106afd824d4bf15344a06b86aa24fda96a8b2 available at: https://gist.github.com/a2eca4289f5f4864cb745f28217f7e33<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-7b5106a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8742-7b5106a.dmg

**Comment by OmegaHeron** [comment]
QA Pass
Tested Ubuntu 16.04.1
Test conditions;
Tested known working domain.  Pass
Tested known not working domain.  Pass

**Comment by sethalves** [comment]
The changes that matter here were dev approved in #8725 (the other changes are just the command-line test client)

**Pull request merged**